bin/ebash

```diff
@@ -119,13 +117,29 @@ else
 
     # Use all arguments to ebash as a command to execute
     if [[ ${#@} -gt 0 ]] ; then
-        # Run the command, but pretend to be inside a try so that die doesn't
-        # print an ugly stack trace.
+
+        # If we were given the path to another script run it inside ebash environment
+        if [[ ${#@} -eq 1 && -e "${1}" ]]; then
+
+            export EBASH EBASH_HOME
+
+            SOURCE="${1}"
+            SCRIPT=$(mktemp --tmpdir ebash-script-XXXXXX)
+            trap_add "rm ${SCRIPT}"
+
+            cp "${SOURCE}" "${SCRIPT}"
+            sed -i -e "s%^#!.*%source ${EBASH}/ebash.sh || { echo \"Unable to source ${EBASH}/ebash.sh\" ; exit 1 ; }%" "${SCRIPT}"
+            chmod +x "${SCRIPT}"
+
+            exec "${SCRIPT}"
```

Comment:
I would have thought the exec would wipe out the env (and therefore the exports) and then either fired the trap or removed it.

Too bad there isn't a way to do this without all the temp file shenanegans

Reply:
Yeah, the temp file wrapper was the simplest thing I could think of. If you think of something better please let me know :-)